src/main/scala/org/bitcoins/core/protocol/blockchain/ChainParams.scala

```diff
@@ -0,0 +1,180 @@
+    * @param nBits An encoded version of the target threshold this block's header hash must be less than or equal to
+    * @param version the block version
+    * @param amount the block reward for the genesis block (50 BTC in Bitcoin)
+    * @return the newly minted genesis block
+    */
+  def createGenesisBlock(timestamp : String, scriptPubKey : ScriptPubKey, time : Long, nonce : Long, nBits : Long,
+                         version : Int, amount : CurrencyUnit) : Block = {
+
+    val timestampHex = timestamp.toCharArray.map(_.toByte)
+    //see https://bitcoin.stackexchange.com/questions/13122/scriptsig-coinbase-structure-of-the-genesis-block
+    //for a full breakdown of the genesis block & its script signature
+    val scriptSignature = ScriptSignature.fromAsm(Seq(BytesToPushOntoStack(4), ScriptNumber(486604799),
+      BytesToPushOntoStack(1), ScriptNumber(4), BytesToPushOntoStack(69), ScriptConstant(timestampHex)))
+    val input = TransactionInput(scriptSignature)
+    val output = TransactionOutput(amount,scriptPubKey)
+    val tx = Transaction(TransactionConstants.version,Seq(input), Seq(output), TransactionConstants.lockTime)
+    val prevBlockHash = DoubleSha256Digest("0000000000000000000000000000000000000000000000000000000000000000")
+    val merkleRootHash = Merkle.computeMerkleRoot(Seq(tx))
+    val genesisBlockHeader = BlockHeader(version,prevBlockHash,merkleRootHash,time,nBits,nonce)
+    val genesisBlock = Block(genesisBlockHeader,CompactSizeUInt(1,1),Seq(tx))
+    genesisBlock
+  }
+}
+
+
+/**
+  * This is the main network parameters
+  */
+object MainNetChainParams extends ChainParams {
+
+  override def networkId = "main"
+
+  override def genesisBlock = createGenesisBlock(1231006505, 2083236893, 0x1d00ffff, 1, Bitcoins(50))
+
+  override def requireStandardTransaction = true
+
+  /**
+    * Base58 prefixes for various keys/hashes on the network
+    * See: https://en.bitcoin.it/wiki/List_of_address_prefixes
+    * @return
+    */
+  override def base58Prefixes : Map[Base58Type,Seq[Byte]] =
+    Map(PubKeyAddress -> BitcoinSUtil.decodeHex("00"),
+      ScriptAddress -> BitcoinSUtil.decodeHex("05"),
+      SecretKey -> BitcoinSUtil.decodeHex("80"),
+      ExtPublicKey -> Seq(BitcoinSUtil.hexToByte("04"), BitcoinSUtil.hexToByte("88"),
+        BitcoinSUtil.hexToByte("b2"), BitcoinSUtil.hexToByte("1e")),
+      ExtSecretKey -> Seq(BitcoinSUtil.hexToByte("04"), BitcoinSUtil.hexToByte("88"),
+        BitcoinSUtil.hexToByte("ad"), BitcoinSUtil.hexToByte("e4")))
+}
+
+object TestNetChainParams extends ChainParams {
+
+  override def networkId = "test"
+
+  override def genesisBlock = createGenesisBlock(1296688602, 414098458, 0x1d00ffff, 1, Bitcoins(50))
+
+  override def requireStandardTransaction = true
+
+  /**
+    * Base58 prefixes for various keys/hashes on the network
+    * See: https://en.bitcoin.it/wiki/List_of_address_prefixes
+    * @return
+    */
+  override def base58Prefixes : Map[Base58Type,Seq[Byte]] = Map(
```

Comment:
Nitpicky: should the "Map" called in the base58Prefixes functions for mainNet/testNet look similar? I.e. for mainNet, it's on next line, but for testNet it's on same line.